### PR TITLE
feat: Apply enableOgtag config set from Dashboard

### PR DESCRIPTION
**Pull request by AhyoungRyu**
### Description Of Changes
Resolves part of https://sendbird.atlassian.net/browse/UIKIT-4085

I applied these two configs to proper places for each. 
- [x] group_channel.channel.enable_ogtag
- [x] open_channel.channel.enable_ogtag

### CODE REVIEW

File: src/modules/OpenChannel/components/OpenChannelMessage/index.tsx [hunk 1]
```diff
@@ -159,7 +160,7 @@ export default function MessagOpenChannelMessageeHoc(props: OpenChannelMessagePr
             }
           })(),
           [MessageTypes.OG]: (() => {
-            if (message?.messageType === 'user') {
+            if (message?.messageType === 'user' && isOgMessageEnabledInOpenChannel) {
```
Comment (AhyoungRyu) [inline src/modules/OpenChannel/components/OpenChannelMessage/index.tsx:163]: It seems this file is the only place that controls ogtag for OpenChannel.

File: src/modules/Thread/components/ThreadList/ThreadListItemContent.tsx [hunk 2]
```diff
@@ -91,6 +93,7 @@ export default function ThreadListItemContent({
   );
   const supposedHoverClassName = supposedHover ? 'sendbird-mouse-hover' : '';
   const isReactionEnabledInChannel = isReactionEnabled && !channel?.isEphemeral;
+  const isOgMessageEnabledInGroupChannel = channel.isGroupChannel() && config.groupChannel.enableOgtag;
```
Comment (AhyoungRyu) [inline src/modules/Thread/components/ThreadList/ThreadListItemContent.tsx:96]: Considering the type of `channel` ([declared at the above](https://github.com/sendbird/sendbird-uikit-react/pull/603/files#diff-8645091ee9cc99dd8dd55fd33017b226611fc495af7fe8d0178757fe00c075f0R36) in this file) is `GroupChannel`, I think checking these two conditions should be enough.

File: src/ui/MessageContent/__tests__/MessageContent.spec.js [hunk 3]
```diff
@@ -43,22 +60,44 @@ const createMockMessage = (process) => {
...
+  })
   // TODO: Add tests after message threading is applied
   // useReplying <-> replyType
   // it('should render components by replyType', () => {});
 
   it('should have class names by own user basic status', () => {
     const className = "test-classname";
     const message = createMockMessage();
-    const { container, queryByTestId } = render(
-      <MessageProvider message={message}>
```
Comment (AhyoungRyu) [inline src/ui/MessageContent/__tests__/MessageContent.spec.js:54]: Removed `<MessageProvider />` in each test suits by mocking `useMessageContext`

File: src/ui/MessageContent/__tests__/MessageContent.spec.js [hunk 4]
```diff
@@ -245,15 +274,61 @@ describe('ui/MessageContent', () => {
...
+        className="classname-for-snapshot"
+        message={message}
+        channel={createMockChannel()}
+        userId="user-id-000"
+      />
     );
     expect(asFragment()).toMatchSnapshot();
   });
+
+  it('should render OGMessageItemBody if config.groupChannel.enableOgtag == true', function () {
```
Comment (AhyoungRyu) [inline src/ui/MessageContent/__tests__/MessageContent.spec.js:287]: Added new ogtag related test cases.

File: src/ui/MessageContent/index.tsx [hunk 5]
```diff
@@ -132,6 +132,8 @@ export default function MessageContent({
   );
   const useReplyingClassName = useReplying ? 'use-quote' : '';
 
+  const isOgMessageEnabledInGroupChannel = channel?.isGroupChannel() && config.groupChannel.enableOgtag;
```
Comment (AhyoungRyu) [inline src/ui/MessageContent/index.tsx:135]: It seems it's another GroupChannel ogtag control part. 

I checked the type of `channel` is `GroupChannel` in [here](
https://github.com/sendbird/sendbird-uikit-react/pull/603/files#diff-0ad613792affc251d9a2e9dbe1135dec1c7ae22cb10f94be24bf9be3c34bbe87R63) but plz tell me if OpenChannel can be used here too.

File: src/utils/index.ts [hunk 6]
```diff
@@ -150,7 +150,7 @@ export const isOGMessage = (message: UserMessage | FileMessage): boolean => !!(
     || message.ogMetaData?.defaultImage
   )
 );
-export const isTextMessage = (message: UserMessage | FileMessage): boolean => isUserMessage(message) && !isOGMessage(message);
```
Comment (AhyoungRyu) [inline src/utils/index.ts:153]: FYI, I removed the dependency between isTextMessage <-> check isOGMessage util function but handled this logic in the components instead.

**Review by sravan-s — APPROVED** [review]

File: src/utils/index.ts (hunk 6 above)
Comment (sravan-s) [reply src/utils/index.ts:153]: Nit/personal-taste (dont need to apply) 
Hmmm, on second thought, better to have handled here.. would be more testable / concise

**Pull request merged**